index.d.ts

```diff
@@ -2,40 +2,57 @@
 
 import { IncomingMessage, ServerResponse } from 'http';
 
-export declare const agent: Agent;
+export default agent;
 
-declare class Agent {
-  middleware: { connect: ConnectMiddlewareFn };
-  lambda: LambdaFn;
-  // logger: Logger // TODO: Is this an official API?
-
-  currentSpan: Span | null;
-  currentTransaction: Transaction | null;
+declare const agent: Agent;
 
+declare class Agent {
+  // Configuration
   start (options: AgentConfigOptions): Agent;
   isStarted (): boolean;
 
-  addFilter (fn: FilterFn): void;
-  addErrorFilter (fn: FilterFn): void;
-  addSpanFilter (fn: FilterFn): void;
-  addTransactionFilter (fn: FilterFn): void;
+  // Data collection hooks
+  middleware: { connect(): ConnectMiddlewareFn };
+  lambda: LambdaFn;
   handleUncaughtExceptions (fn: UncaughtExceptionFn): void;
 
-  captureError (err: Error | string | ParameterizedMessageObject, options?: CaptureErrorOptions, callback?: CaptureErrorCallback): void;
-
+  // Errors
+  captureError (
+    err: Error | string | ParameterizedMessageObject,
+    callback?: CaptureErrorCallback
+  ): void;
+  captureError (
+    err: Error | string | ParameterizedMessageObject,
+    options?: CaptureErrorOptions,
+    callback?: CaptureErrorCallback
+  ): void;
+
+  // Transactions
   startTransaction (name?: string, type?: string, options?: TransactionOptions): Transaction | null;
   setTransactionName (name: string): void;
   endTransaction: EndTransactionFn;
-  
+  currentTransaction: Transaction | null;
+
+  // Spans
   startSpan: StartSpanFn;
+  currentSpan: Span | null;
 
+  // Context
   setTag: SetTagFn; // TODO: Is this how to add a declared function to a class?
   addTags: AddTagsFn;
   setUserContext (user: UserObject): void;
   setCustomContext (custom: object): void;
 
+  // Transport
+  addFilter (fn: FilterFn): void;
+  addErrorFilter (fn: FilterFn): void;
+  addSpanFilter (fn: FilterFn): void;
+  addTransactionFilter (fn: FilterFn): void;
   flush (callback: Function): void;
   destroy (): void;
+
+  // Utils
+  logger: Logger; // TODO: Should we advertise this API?
```

Comment:
It's been there for a long time, and I don't expect we'll move it any time soon, so why not? 🤷‍♂️

Reply:
I just wasn't sure what the user-case from a users point of view to use this was and it's not documented. It's more of an internal thing. But you're right, that if we ever wanted to remove it we probably should make it a breaking change anyway, so in that case we might as well add it to the typings file